src/languages/index.ts

```diff
@@ -0,0 +1,76 @@
+import lang from 'i18n-js';
+import english from './_english.json';
+import french from './_french.json';
+
+import { simpleObjectProxy } from '@/languages/utils';
+
+/**
+ * Use English as our "template" for translations. All other translations
+ * should match the objects and keys within the English translation.
+ */
+export type Translation = typeof english;
+
+export enum Language {
+  English = 'en',
+  French = 'fr',
+}
+
+export const resources: {
+  [key in Language]: Translation;
+} = {
+  en: english,
+  // @ts-ignore
+  fr: french,
+};
+
+export const supportedLanguages: {
+  [key in Language]: string;
+} = {
+  en: 'English',
+  fr: 'French',
+};
+
+// Configure languages
+lang.defaultLocale = Language.English;
+lang.locale = Language.English;
+lang.fallbacks = true;
+lang.translations = Object.assign(
+  {},
+  ...Object.keys(resources).map(key => ({
+    [key]: resources[key as Language].translation,
+  }))
+);
+
+export const updateLanguageLocale = (code: Language) => {
```

Comment:
Adding this type required type-casting in other files, but I think it's still worth doing because it forces the developer to ensure they're passing the right values. And in the future maybe we can include the enum values in state mgmt so that we don't need to type-cast.